x-pack/legacy/plugins/apm/server/lib/helpers/setup_request.ts

```diff
@@ -5,26 +5,31 @@
  */
 
 import { Legacy } from 'kibana';
-import { Server } from 'hapi';
 import moment from 'moment';
 import { getESClient } from './es_client';
 import { getUiFiltersES } from './convert_ui_filters/get_ui_filters_es';
 import { PromiseReturnType } from '../../../typings/common';
 import { getApmIndices } from '../settings/apm_indices/get_apm_indices';
+import { StaticIndexPattern } from '../../../../../../../src/legacy/core_plugins/data/public';
+import { getDynamicIndexPattern } from '../index_pattern/get_dynamic_index_pattern';
 
-function decodeUiFilters(server: Server, uiFiltersEncoded?: string) {
-  if (!uiFiltersEncoded) {
+function decodeUiFilters(
+  indexPattern: StaticIndexPattern | undefined,
+  uiFiltersEncoded?: string
+) {
+  if (!uiFiltersEncoded || !indexPattern) {
     return [];
   }
   const uiFilters = JSON.parse(uiFiltersEncoded);
-  return getUiFiltersES(server, uiFilters);
+  return getUiFiltersES(indexPattern, uiFilters);
 }
 
 export interface APMRequestQuery {
   _debug?: string;
   start?: string;
   end?: string;
   uiFilters?: string;
+  processorEvent?: 'transaction' | 'error' | 'metric';
```

Comment:
Probably at some point this should be a constant or enum, as we're using it in several places

Reply:
Good idea. For now I've added it to `x-pack/legacy/plugins/apm/typings/common.d.ts`. 
LMK if you have ideas for a better location for this.

Reply:
Perhaps in `common/events.ts`? Adding it in **/typings/* makes it globally available for all plugins, and I think a constant or enum would add more value than a type because we can use it at runtime.